janusgraph-server/src/main/java/org/janusgraph/graphdb/server/JanusGraphSettings.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.server;
+
+import org.apache.tinkerpop.gremlin.server.Settings;
+import org.yaml.snakeyaml.TypeDescription;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Objects;
+
+public class JanusGraphSettings extends Settings {
+    public JanusGraphSettings.GrpcServerSettings grpcServer = new JanusGraphSettings.GrpcServerSettings();
+
+    public static class GrpcServerSettings {
+        public boolean enable = true;
+        public int port = 10182;
+    }
+
+    public static JanusGraphSettings read(final String file) throws Exception {
+        InputStream input = new FileInputStream(new File(file));
+        return read(input);
+    }
+
+    public static JanusGraphSettings read(final InputStream stream) {
```

Comment:
I find it very error-prone to just copy&paste the code from the TinkerPop `Settings` class here. Especially because SnakeYaml uses Reflection, the compiler will not instantly notice a change in the TinkerPop class and therefore resulting incompatibilities.
Would it be possible to call `super.read(...)` and somehow integrate the obtained `Settings` object into the `JanusGraphSettings`? This isn't a nice solution either but at least it would not require to duplicate code.

Reply:
Yaml fails to parse, if anything is not correctly defined.

Reply:
We are now using the default gremlin settings parser.